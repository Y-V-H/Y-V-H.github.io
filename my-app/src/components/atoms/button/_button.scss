@import '../../_settings/variables';

.a-btn {
    font-family: $montserrat-font;
    text-transform: uppercase;
    font-size: 16px;
    line-height: 20px;
    border: none;
    cursor: pointer;
    border-radius: 5px;
}

.a-btn__primary {
    background: rgba($granite-gray, 0.68);
    padding: 12px 18px;
    color: $infra-red;
}

.a-btn__secondary {
    background: $infra-red;
    padding: 18px 24px;
    color: $white;
}

.a-btn__tertiary {
    background: $black;
}