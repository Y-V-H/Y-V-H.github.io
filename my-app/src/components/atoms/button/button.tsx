import React from 'react';
import { clsx } from 'clsx';

interface ButtonProps {
    children?: string;
    className?: string;
    onClick?: React.MouseEventHandler<HTMLButtonElement>;
    type?: 'submit' | 'reset' | 'button';
}

export const Button = ({
    children,
    className,
    onClick,
    type
}: ButtonProps) => {
    return (
        <button
            className={clsx(`a-btn ${className}`)}
            onClick={onClick}
            type={type || 'button'}
        >
            {children}
        </button>
    )
}