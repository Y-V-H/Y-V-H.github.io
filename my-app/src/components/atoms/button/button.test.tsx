// /** @jest-environment jsdom */

import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Button } from './button';
import { render, screen } from '@testing-library/react';

test('should render a button', () => {
    const component = renderer.create(
        <Button />
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test('should render a button', () => {
    render(<Button />)
})

it('should render a button with the class of primary', () => {
    render(<Button className='a-btn__primary'/>);
    const primaryButton = screen.getByRole('button');
    expect(primaryButton).toHaveClass('a-btn__primary');
})

