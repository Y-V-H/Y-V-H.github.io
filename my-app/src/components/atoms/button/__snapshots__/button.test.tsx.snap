// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a button 1`] = `
<button
  className="a-btn undefined"
  type="button"
/>
`;
